include/cocaine/rpc/asio/decoder.hpp

```diff
@@ -49,6 +49,29 @@ struct decoded_message_t {
         return object.via.array.ptr[2];
     }
 
+    auto
+    trace_id() const ->uint64_t {
+        if(object.via.array.size < 4) {
+            return 0;
```

Comment:
0 means "no trace". Anyway if you take a look at "headers" (cocaine/cocaine-core#203) PR you'll find that there is boost::optional there